Build fails with Clang 7.0 due to local variable copies

When building with Clang 7.0, I'm getting thousands of errors like this.
```
cbmc/src/pointer-analysis/value_set_fivr.cpp:360:10: error: local variable 'od' will be copied despite being returned by name [-Werror,-Wreturn-std-move]
  return od;
         ^~
/home/kareem/doc/cbmc/src/pointer-analysis/value_set_fivr.cpp:360:10: note: call 'std::move' explicitly to avoid copying
  return od;
         ^~
         std::move(od)
```
I'll fix these soon. Here's the environment:
```
> clang --version
clang version 7.0.0 (tags/RELEASE_700/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

--- a/jbmc/src/java_bytecode/java_string_library_preprocess.cpp
+++ b/jbmc/src/java_bytecode/java_string_library_preprocess.cpp
@@ -1171,7 +1171,8 @@ codet java_string_library_preprocesst::make_assign_function_from_call(
 /// \return An expression contaning a symbol `tmp_type_name` where `type_name`
 ///         is the given argument (ie. boolean, char etc.). Which represents the
 ///         primitive value contained in the given object.
-exprt java_string_library_preprocesst::get_primitive_value_of_object(
+optionalt<symbol_exprt>
+java_string_library_preprocesst::get_primitive_value_of_object(
   const exprt &object,
   irep_idt type_name,
   const source_locationt &loc,
@@ -1222,7 +1223,7 @@ exprt java_string_library_preprocesst::get_primitive_value_of_object(
     object_type=symbol_typet("java::java.lang.Double");
   }
   else if(type_name==ID_void)
-    return nil_exprt();
+    return {};
   else
     UNREACHABLE;
 
@@ -1398,9 +1399,12 @@ exprt java_string_library_preprocesst::make_argument_for_format(
     }
     else if(name==ID_int || name==ID_float || name==ID_char || name==ID_boolean)
     {
-      exprt value=get_primitive_value_of_object(
+      const auto value = get_primitive_value_of_object(
         arg_i, name, loc, symbol_table, code_not_null);
-      code_not_null.add(code_assignt(field_expr, value), loc);
+      if(value.has_value())
+        code_not_null.add(code_assignt(field_expr, *value), loc);
+      else
+        code_not_null.add(code_assignt(field_expr, nil_exprt()), loc);
     }
     else
     {

--- a/jbmc/src/java_bytecode/java_string_library_preprocess.h
+++ b/jbmc/src/java_bytecode/java_string_library_preprocess.h
@@ -21,6 +21,7 @@ Date:   March 2017
 #include <util/string_expr.h>
 
 #include <util/ieee_float.h> // should get rid of this
+#include <util/optional.h>
 
 #include <array>
 #include <unordered_set>
@@ -324,7 +325,7 @@ private:
     symbol_table_baset &symbol_table,
     code_blockt &code);
 
-  exprt get_primitive_value_of_object(
+  optionalt<symbol_exprt> get_primitive_value_of_object(
     const exprt &object,
     irep_idt type_name,
     const source_locationt &loc,

--- a/src/ansi-c/literals/convert_float_literal.cpp
+++ b/src/ansi-c/literals/convert_float_literal.cpp
@@ -101,5 +101,5 @@ exprt convert_float_literal(const std::string &src)
     return complex_exprt(zero_real_component, result, complex_type);
   }
 
-  return result;
+  return std::move(result);
 }

